docs/architecture/cloud-native/azure-caching.md

```diff
@@ -0,0 +1,60 @@
+---
+title: Caching in a cloud-native application
+description: Learn about caching strategies in a cloud-native application.
+author: robvet
+ms.date: 12/28/2019
+---
+# Caching in a cloud-native app
+
+[!INCLUDE [book-preview](../../../includes/book-preview.md)]
+
+The benefits of caching are well understood. The technique works by temporarily copying frequently accessed data from a backend data store to *fast storage* that's located closer to the application. Caching is often implemented where...
+
+- Data remains relatively static.
+- Data access is slow, especially compared to the speed of the cache.
+- Data is subject to high levels of contention.
+
+## Why?
+
+As discussed in the [Microsoft caching guidance](https://docs.microsoft.com/azure/architecture/best-practices/caching?toc=%2Fazure%2Fredis-cache%2Ftoc.json), caching can increase performance, scalability, and availability for individual microservices and the system as a whole. It reduces the latency and contention of handling large volumes of concurrent requests to a data store. As data volume and the number of users increase, the greater the benefits of caching become.
+
+Caching is most effective when a client repeatedly reads data that is immutable or that changes infrequently. Examples include reference information such as product and pricing information, or shared static resources that are costly to construct.
+
+While microservices should be stateless, a distributed cache can support concurrent access to session state data when absolutely required.
+
+Also consider caching to avoid repetitive computations. If an operation transforms data or performs a complicated calculation, cache the result for subsequent requests.
+
+## Caching architecture
+
+Cloud native applications typically implement a distributed caching architecture. The cache is hosted as a [cloud-based backing service](./definition.md#backing-services), separate from the microservices. Figure 5-20 shows the architecture.
+
+![Caching in a cloud native app](media/caching-in-a-cloud-native-app.png)
+
+**Figure 5-20**: Caching in a cloud native app
+
+In the figure above, note how the cache is independent of and shared by the microservices. In this scenario, the cache is invoked by the [API Gateway](./front-end-communication.md#Front-end-client-communication). As discussed in chapter 4, the gateway serves as a front end for all incoming requests. The distributed cache increases system responsiveness by returning cached data whenever possible. Additionally, separating the cache from the services allows the cache to scale up or out independently to meet increased traffic demands.
```

Comment:
figure above -> previous figure
bad anchor -> did you really mean to link to the article or were you trying to link to a section?

```suggestion
In the previous figure, note how the cache is independent of and shared by the microservices. In this scenario, the cache is invoked by the [API Gateway](./front-end-communication.md). As discussed in chapter 4, the gateway serves as a front end for all incoming requests. The distributed cache increases system responsiveness by returning cached data whenever possible. Additionally, separating the cache from the services allows the cache to scale up or out independently to meet increased traffic demands.
```